### PR TITLE
[DataGrid] Fix `Maximum call stack size exceeded` error when using fractional width

### PR DESCRIPTION
**Pull request by cherniavskii**
- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui/mui-x/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

Fixes https://github.com/mui/mui-x/issues/9510

Before: https://codesandbox.io/s/crimson-bash-7m8g6c?file=/demo.tsx
After: https://codesandbox.io/s/sad-pine-6c2j29?file=/demo.tsx

### CODE REVIEW

File: packages/grid/x-data-grid/src/hooks/features/dimensions/useGridDimensions.ts [hunk 1]
```diff
@@ -122,7 +122,7 @@ export function useGridDimensions(
       };
 
       const scrollInformation = hasScroll({
-        content: { width: Math.round(columnsTotalWidth), height: rowsMeta.currentPageTotalHeight },
+        content: { width: columnsTotalWidth, height: rowsMeta.currentPageTotalHeight },
```
Comment (cherniavskii) [inline packages/grid/x-data-grid/src/hooks/features/dimensions/useGridDimensions.ts:125]: In https://github.com/mui/mui-x/pull/9236 we started to use fractional size. Rounding the columns total width breaks the equality comparison

**Comment by AnastasiaEvgenia** [comment]
Hi.
This error still exists in version 6.9.1.
This seems to be happening sometimes if you start resizing your grid.

Sand box
https://codesandbox.io/s/maximum-call-stack-size-exceeded-zhz6dw?file=/package.json
Untitled video - Made with Clipchamp
Screenshot_1733
Screenshot_1732

**Comment by cherniavskii** [comment]
Thanks @AnastasiaEvgenia 
It should be fixed by https://github.com/mui/mui-x/pull/9554

**Comment by seemX17** [comment]
Fixed with latest `version 6.12.1` :) Was facing similar issue

Screen Shot 2023-09-06 at 11 57 10 AM

**Comment by gaisdav** [comment]
same error in "@mui/x-data-grid-pro": "7.1.1", does anybody know how to fix?

**Comment by cherniavskii** [comment]
@gaisdav Are you sure it's the "Maximum call stack size exceeded" error?
There was a similar (yet different) issue reported in https://github.com/mui/mui-x/issues/12851 and [fixed in 7.3.1](https://github.com/mui/mui-x/issues/12851#issuecomment-2102312584)

If the issue linked above is unrelated to the issue you're experiencing, please open a new issue with a minimal reproduction example. Thanks!

**Comment by gaisdav** [comment]
@cherniavskii yes, let me show the error. 

Screenshot 2024-05-09 at 18 37 24

**Comment by cherniavskii** [comment]
@gaisdav Hmm, this looks like https://github.com/mui/mui-x/issues/11483#issuecomment-2081123128
Can you create a reproduction example and share it in https://github.com/mui/mui-x/issues/11483?

**Comment by gaisdav** [comment]
@cherniavskii 
I'll try, thanks!

**Pull request merged**